### PR TITLE
Use msg id's from nsdl handle to verify registration related callbacks

### PR DESCRIPTION
**Pull request by anttiylitokola**

### CODE REVIEW

**Comment by anttiylitokola** [comment]
@yogpan01

File: source/m2mnsdlinterface.cpp [hunk 1]
```diff
@@ -438,17 +428,16 @@ uint8_t M2MNsdlInterface::received_from_server_callback(struct nsdl_s * /*nsdl_h
...
             if(coap_header->msg_code == COAP_MSG_CODE_RESPONSE_CHANGED) {
                 tr_debug("M2MNsdlInterface::received_from_server_callback - registration_updated successfully");
                 _observer.registration_updated(*_server);
             } else {
                 tr_error("M2MNsdlInterface::received_from_server_callback - registration_updated failed %d", coap_header->msg_code);
                 _registration_timer->stop_timer();
-                _register_id = -1;
-                _register_id = sn_nsdl_register_endpoint(_nsdl_handle,_endpoint);
+                sn_nsdl_register_endpoint(_nsdl_handle,_endpoint);
+                _register_ongoing = true;
```
Comment (yogpan01) [inline source/m2mnsdlinterface.cpp]: Should this flag be before the sn_nsdl_register_endpoint() ?

File: source/m2mnsdlinterface.cpp [hunk 2]
```diff
@@ -289,19 +286,15 @@ bool M2MNsdlInterface::send_update_registration(const uint32_t lifetime)
...
+                                                  _endpoint->lifetime_ptr,
+                                                  _endpoint->lifetime_len) != 0;
         }
     } else {
         if(_nsdl_handle) {
-            _update_id = -1;
-            _update_id = sn_nsdl_update_registration(_nsdl_handle, NULL, 0);
-            tr_debug("M2MNsdlInterface::send_update_registration - regular update- _update_id %" PRId32, _update_id);
-            success = _update_id != 0;
+            tr_debug("M2MNsdlInterface::send_update_registration - regular update");
```
Comment (yogpan01) [inline source/m2mnsdlinterface.cpp]: Where is update_register_ongoing flag is set to true ?

File: source/m2mnsdlinterface.cpp [hunk 3]
```diff
@@ -310,18 +303,14 @@ bool M2MNsdlInterface::send_update_registration(const uint32_t lifetime)
...
     bool success = false;
-    //Does not clean resources automatically
-    if(_unregister_id == 0) {
-       _unregister_id = -1;
-       _unregister_id = sn_nsdl_unregister_endpoint(_nsdl_handle);
-       tr_debug("M2MNsdlInterface::send_unregister_message - unregister_id %" PRId32, _unregister_id);
-       success = _unregister_id != 0;
-    }
+    success = sn_nsdl_unregister_endpoint(_nsdl_handle) != 0;
+    _unregister_ongoing = true;
```
Comment (yogpan01) [inline source/m2mnsdlinterface.cpp]: Should this flag be set to true before calling sn_nsdl_unregister_endpoint() ?

**Comment by anttiylitokola** [comment]
rebuild

**Pull request merged**